### DISCUSSION TITLE
Can someone help me with my config file?

### DISCUSSION THREAD

**Discussion by LeonoraSewidan** [Ulm-IQO/qudi-core#94] [Q&A] [answered]
Hello, 
I wanted to start using qudi because it was recommended to me during my project, but I have little to no programming experience. I am a little overwhelmed and even though I was working with a lab using qudi, the person who set it up is no longer available and he used the old qudi version. 
I have downloaded qudi-core and the  iqo-modules. Now I wanted to do something as simple as add my DAQ card in the configuration file (I have the NI-6363), but I am already failing while doing so. I am not quite sure what to do. In the iqo-modules hardware folder is a folder for the ni_x_series and in those files is an example for nicard config, but in the end I am still not sure how to make it work. 

So far, I have only connected my piezo controller to the DAQ card on the analog analog channels AO 0 to  AO3, and I only copied the example config from the ni_x_series_analog_output.py file, which I thought would be enough for now since I don't have a photon signal yet, but in the other ni_x_series files are more example configs. 
If someone is willing to help me, that would be great because I am lost and don't know how to make it work. After copying the first example config, I adjusted in the default config file the scanning_probe_logic to use the nicard scanner instead of the scanner_dummy and tried to run it to see if it will work. But I got this error: 
No module class "NIXSeriesAnalogOutput" found in module "qudi.hardware.ni_x_series.ni_x_series_in_streamer"

I appreciate any help how to set up the config file. 
Thank you

**Comment by timoML** [comment]
Can you post the nicard related parts of your config file? It's probably only a typo in the module.Class parameters that you're specifiying.

**Reply by LeonoraSewidan** [reply]
So far I literally just copied it out of the specified file, so: 
    nicard_63XX_ao:
        module.Class: 'ni_x_series.ni_x_series_in_streamer.NIXSeriesAnalogOutput'
        options:
            device_name: 'Dev1'
            channels:
                ao0:
                    limits: [-10.0, 10.0]
                    keep_value: True
                ao1:
                    limits: [-10.0, 10.0]
                    keep_value: True
                ao2:
                    limits: [-10.0, 10.0]
                    keep_value: True
                ao3:
                    limits: [-10.0, 10.0]
                    keep_value: True

**Reply by timoML** [reply]
It seems like the dosctring is broken. Can you please try:

`module.Class: 'ni_x_series.ni_x_series_analog_output.NIXSeriesAnalogOutput'`

**Reply by LeonoraSewidan** [reply]
Thank you, that worked to at least simply load the hardware. But it seems it is not as simple as I thought, I thought I can simply change in the default config file the logic of the scanning_probe_logic from the scanner dummy to my nicard and run the scanner_gui, but this does not work. 
<qudi.hardware.ni_x_series.ni_x_series_analog_output.NIXSeriesAnalogOutput(0x166965404e0) at 0x0000016699A990C0>" connected to connector "scanner" does not implement interface "ScanningProbeInterface"

**Reply by timoML** [reply, marked as answer]
The scanning toolchain got a bit more complicated to configure. Did you see this [readme ](https://github.com/Ulm-IQO/qudi-iqo-modules/blob/main/docs/setup_confocal_scanning.md)document? It shows how to connect the scanning_logic to a nicard based ni_scanner.

**Reply by LeonoraSewidan** [reply]
No, I did not see that, thank you. I will look at it and try it and come back if I run into further questions.

**Reply by timoML** [reply]
No worries, thanks for pointing us to this broken docstring. Can you mark the question answered, if you managed to configure your scanning toolchain?

**Reply by LeonoraSewidan** [reply]
Thank you, the confocal scan and the time series is running now with this configuration from the readme document, but I noticed that the countrate doesnt match up if I run the time series counter (10k counts) and a time tagger counter (20k counts). I thought this might have to do with the trigger level (so basically the voltage) but I cannot seem to find a trigger level in the time series logic and in the ni_instreamer hardware module I also cannot find a trigger level. The count behaviour is the same, just the actual countrate is different. Is there some value that  I am overlooking?

**Reply by timoML** [reply]
Most of the nicards don't allow to configure their counter trigger levels. This is why qudi is also not implementing it.

However, doubling of the countrate makes me suspicious. You might have reflections, eg. by a impadance mismatch, in your cabling. The easiest way to check is probably to connect a signal generator as a source (replacing the APD). Then you'll know the expected count rate exactly and will be able to identify which device is potentially counting reflected TTL pulses.

**Reply by LeonoraSewidan** [reply]
Thank you for the info, that is a good tip and will try it, but I am pretty sure the time tagger is correct, since the cable that is connected to the DAQ card was previously connected to an analog counter which gave me the same result as the time tagger.
